src/main/java/org/myrobotlab/service/MouthControl.java

```diff
@@ -23,27 +29,72 @@
 
   private static final long serialVersionUID = 1L;
   public final static Logger log = LoggerFactory.getLogger(MouthControl.class);
+  @Deprecated /* future releases members will be protected use appropriate methods instead */
   public int mouthClosedPos = 20;
+  @Deprecated /* future releases members will be protected use appropriate methods instead */
   public int mouthOpenedPos = 4;
+  @Deprecated /* future releases members will be protected use appropriate methods instead */
   public int delaytime = 75;
+  @Deprecated /* future releases members will be protected use appropriate methods instead */
   public int delaytimestop = 150;
+  @Deprecated /* future releases members will be protected use appropriate methods instead */
   public int delaytimeletter = 45;
+  
+  protected final Set<String> servoServices = new HashSet<>();
+  protected final Set<String> speechServices = new HashSet<>();
+
   transient private ServoControl jaw;
   transient private SpeechSynthesis mouth;
+  protected String jawName;
+  protected String mouthName;
 
   public MouthControl(String n, String id) {
     super(n, id);
+    registerForInterfaceChange(ServoControl.class);

```

Comment:
I feel like all services should register for interfaces they are interested in.  so, the service should have a method that retuns back a list of interface classes ...  when runtime creates the service... it would call that method and register the callbacks automatically.

Reply:
it currently automatically creates callbacks and invokes them when appropriate events are triggered

Reply:
if runtime is handling this , we should remove the registrations here